arkouda/timeclass.py

```diff
@@ -144,6 +143,158 @@ def __init__(self, array, unit: str = _BASE_UNIT):  # type: ignore
+            self._day += 2
+            self._day[d >= 32] -= 31
+
+            # Day of year
+            month_days = [
+                [0, 0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334],
+                [0, 0, 31, 60, 91, 121, 152, 182, 213, 244, 274, 305, 335],
+            ]
+
+            lyear1 = self._year % 4 == 0
+            lyear2 = self._year % 100 != 0
+            lyear3 = self._year % 400 == 0
+            lyear = lyear1 & lyear2
+            lyear = cast((lyear | lyear3), int64)
+
+            days = self._day[:]
+            self._date = []
+            for i in range(days.size):
+                if self._month[i] == 2 and self._day[i] == 29 and lyear[i] == 0:
+                    self._month[i] += 1
+                    self._day[i] = 1
+                    days[i] += month_days[lyear[i]][self._month[i] - 1]
+                else:
+                    days[i] += month_days[lyear[i]][self._month[i]]
+                self._date.append(f"{self._year[i]}-{self._month[i]:02d}-{self._day[i]:02d}")
+            self._dayofyear = cast(ak.array(days), int64)
+
+            # Day of week
+            self._dayofweek = cast(((self.values / _get_factor("d")) + 3), int64)
+            self._dayofweek %= 7
+
+            # Week of year
+            week = (self._dayofyear - 1) - self._dayofweek + 3
+            week[week >= 0] //= 7
+            week[week >= 0] += 1
+
+            for i in range(days.size):
+                if week[i] < 0:
+                    year = int(self._year[i] - 1)
+                    if week[i] > -2 or (week[i] == -2 and _isLeapYear(year)):
+                        week[i] = 53
+                    else:
+                        week[i] = 52
+                elif week[i] == 53:
+                    if 31 - self._day[i] + self.dayofweek[i] < 3:
+                        week[i] = 1
+            self._weekofyear = week
+
+    # Datetime values
+    @property
+    def date(self):
+        return self._date
+
+    @property
+    def year(self):
+        return self._year
+
+    @property
+    def month(self):
+        return self._month
+
+    @property
+    def day(self):
+        return self._day
+
+    @property
+    def dayofyear(self):
+        return self._dayofyear
+
+    @property
+    def dayofweek(self):
+        return self._dayofweek
+
+    @property
+    def week(self):
+        return self._weekofyear
+
+    day_of_week = dayofweek
+    weekday = dayofweek
+    day_of_year = dayofyear
+    week_of_year = week
+    weekofyear = week
+
+    @property
+    def hour(self):
+        return self._hour
+
+    @property
+    def minute(self):
+        return self._minute
+
+    @property
+    def second(self):
+        return self._second
+
+    @property
+    def total_seconds(self):
+        ts = cast(self.values, float)
+        return ts / _get_factor("s")
+
+    # Timedelta values
+    @property
+    def components(self):
+        dt = self.to_pandas()
+        return dt.components
```

Comment:
should we be doing `self.to_pandas` here? I feel like we should just modify this function to return our components in a similar format as pandas